zulip_bots/zulip_bots/lib.py

```diff
@@ -197,6 +205,12 @@ def handle_message(message):
             if message['content'] is None:
                 return
 
+        # Handle empty message directly if bot (handler_class) does not
+        # explicitly request to do so
+        if len(message['content']) == 0 and handle_empty_here:
+            restricted_client.send_reply(message, default_empty_message_response)
+            return
```

Comment:
a) It doesn't seem to ;)
b) This return is from `handle_message()`, which is called by the lib for each message;
c) A few lines above, it returns, and not for exceptional circumstances, which indicates it should be fine.